akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -153,19 +156,33 @@ private[akka] final case class MonitorInterceptor[T: ClassTag](actorRef: ActorRe
 private[akka] final class LogMessagesInterceptor(val opts: LogOptions) extends BehaviorInterceptor[Any, Any] {
 
   import BehaviorInterceptor._
+  import LogMessagesInterceptor._
+
+  private val logger = opts.getLogger.orElse(LoggerFactory.getLogger(getClass))
 
   override def aroundReceive(ctx: TypedActorContext[Any], msg: Any, target: ReceiveTarget[Any]): Behavior[Any] = {
-    if (opts.enabled)
-      opts.logger.getOrElse(ctx.asScala.log).log(opts.level, "received message {}", msg)
+    log(logMessageTemplate, msg, ctx)
     target(ctx, msg)
   }
 
   override def aroundSignal(ctx: TypedActorContext[Any], signal: Signal, target: SignalTarget[Any]): Behavior[Any] = {
-    if (opts.enabled)
-      opts.logger.getOrElse(ctx.asScala.log).log(opts.level, "received signal {}", signal)
+    log(logSignalTemplate, signal, ctx)
     target(ctx, signal)
   }
 
+  private def log(template: String, messageOrSignal: Any, context: TypedActorContext[Any]): Unit = {
+    if (opts.enabled) {
+      opts.level match {
+        case Level.ERROR => logger.error(template, context.asScala.self.path, messageOrSignal)
+        case Level.WARN  => logger.warn(template, context.asScala.self.path, messageOrSignal)
+        case Level.INFO  => logger.info(template, context.asScala.self.path, messageOrSignal)
+        case Level.DEBUG => logger.debug(template, context.asScala.self.path, messageOrSignal)
+        case Level.TRACE => logger.trace(template, context.asScala.self.path, messageOrSignal)
+        case _           => logger.debug(template, context.asScala.self.path, messageOrSignal)
```

Comment:
How can it be anything else here when `Level` is an enum??

Reply:
can't, but makes it somewhat future proof if they add another enum value

maybe we would rather detect and throw

Reply:
Sounds better to me, also extremely unlikely that a new logging level would be introduced

Reply:
fixed in 2c85f07ff16b759fc35aebee563ec2975c623720